### ISSUE TITLE
Incorrect peer dependency warning with "@types/react@^16.9.23||^17" and "react@^16.4.2||^17"

### ISSUE THREAD

**Issue by JinZihang** [Nozbe/WatermelonDB#1404]
Hi, I am building a react native project.I tried to follow the installation steps from [this page](https://nozbe.github.io/WatermelonDB/Installation.html). When I do `yarn add @nozbe/watermelondb`, I got the warning as follows:

```
warning "@nozbe/watermelondb > @nozbe/with-observables@1.4.0" has incorrect peer dependency "@types/react@^16.9.23||^17".
warning "@nozbe/watermelondb > @nozbe/with-observables@1.4.0" has incorrect peer dependency "react@^16.4.2||^17".
```

May I know if there will be any issue if I just ignore this warning and proceed? Or should I downgrade the above dependencies and all related ones? Or should I just wait until watermelonDB is compatible?

Below is my package.json file.

```
{
  "name": "DreamProject",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "start": "react-native start",
    "test": "jest",
    "lint": "eslint ."
  },
  "dependencies": {
    "@nozbe/watermelondb": "^0.24.0",
    "@react-navigation/bottom-tabs": "^6.4.0",
    "@react-navigation/native": "^6.0.13",
    "@react-navigation/native-stack": "^6.9.0",
    "@types/hoist-non-react-statics": "^3.3.1",
    "react": "18.1.0",
    "react-native": "0.70.2",
    "react-native-safe-area-context": "^4.4.1",
    "react-native-screens": "^3.18.0",
    "react-native-vector-icons": "^9.2.0"
  },
  "devDependencies": {
    "@babel/core": "^7.12.9",
    "@babel/plugin-proposal-decorators": "^7.19.3",
    "@babel/preset-env": "^7.19.3",
    "@babel/runtime": "^7.12.5",
    "@react-native-community/eslint-config": "^2.0.0",
    "@types/jest": "^29.1.1",
    "@types/react": "^18.0.21",
    "@types/react-native": "^0.70.4",
    "@types/react-native-vector-icons": "^6.4.12",
    "@types/react-test-renderer": "^18.0.0",
    "babel-jest": "^26.6.3",
    "eslint": "^7.32.0",
    "jest": "^26.6.3",
    "metro-react-native-babel-preset": "0.72.3",
    "react-test-renderer": "18.1.0",
    "typescript": "^4.8.4"
  },
  "jest": {
    "preset": "react-native"
  }
}
```

Thank you.

**Comment by wtcosta** [comment]
Mesma duvida aqui

**Comment by radex** [comment]
Fixed in @nozbe/with-observables@1.5.0-2 / @nozbe/watermelondb@0.26.0-9
